### PR TITLE
backport: Merge bitcoin#18814, 18781

### PR DESCRIPTION
**Pull request by vijaydasmp**
## Issue being fixed or feature implemented
bitcoin backports

## What was done?
Backported bitcoin changes

## How Has This Been Tested?
CI Run

### CODE REVIEW

**Review by knst — CHANGES_REQUESTED** [review]

File: doc/release-notes.md [hunk 1]
```diff
@@ -45,6 +45,11 @@ This will be used to support custom builds such as for nightly builds with extra
 
 Additionally, the Guix system will now produce debug symbols for MacOS. 
 
+## Backports from Bitcoin Core
```
[the remark this answers is not in the document]
Comment (vijaydasmp) [reply doc/release-notes.md:48]: sure

**Comment by vijaydasmp** [comment]
Hello @UdjinM6, @PastaPastaPasta, requesting review

File: doc/release-notes-19202.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+- The `-debug=db` logging category, which was deprecated in 0.20 and replaced by
```
Comment (knst) [inline doc/release-notes-19202.md:1]: missing lines; need to specify category what this changes are for:

```suggestion
+Updated settings
----------------

- The `-debug=db` logging category, which was deprecated in 0.20 and replaced by
```

Comment (vijaydasmp) [reply doc/release-notes-19202.md:4]: Updated

**Review by knst — APPROVED** [review]
utACK

**Review by UdjinM6 — APPROVED** [review]
utACK

**Comment by PastaPastaPasta** [comment]
I think 19202 is technically a breaking change and should be delayed until v21

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]
includes breaking changes; and we're currently merging into v20.1

**Review by knst — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Pull request merged**